lib/business/janitor.dart

```diff
@@ -0,0 +1,73 @@
+import '../models/tasks.dart';
+import '../utils/io.dart';
+
+class Janitor {
+  late List<JanitorTask> janitorTasks;
+
+  Janitor() {
+    refresh();
+  }
+
+  Future<bool> editStatus(TaskStatus status, int taskIndex) async {
+    if (taskIndex >= janitorTasks.length || taskIndex < 0) return false;
+
+    janitorTasks[taskIndex].status = status;
+
+    var io = IO();
+    var parameter = <String, String>{'request_id': taskIndex.toString()};
+    await io.patchJanitor(parameter, status);
+
+    return true;
+  }
+
+  Future<bool> addTask(JanitorTask task) async {
+    janitorTasks.add(task);
+
+    var io = IO();
+    await io.postJanitor(null, task);
+
+    return true;
+  }
+
+  Future<bool> deleteTask(int taskIndex) async {
+    if (taskIndex >= janitorTasks.length || taskIndex < 0) return false;
+
+    janitorTasks.removeAt(taskIndex);
+
+    var io = IO();
+    var parameter = <String, String>{'request_id': janitorTasks[taskIndex].uid};
+    await io.deleteJanitor(parameter);
+
+    return true;
+  }
+
+  Future<void> refresh() async {
+    var io = IO();
+    janitorTasks = await io.getJanitor(null);
+  }
+
+  List<JanitorTask> filter(List<String> statuses, List<String> roomIDs) {
```

Comment:
A státuszokat érdemes a státusz enum típus változóiként vizsgálni. Maga a JanitorTask osztály is ilyen tagváltozó típussal tárolja őket.
```suggestion
  List<JanitorTask> filter(List<TaskStatus> statuses, List<String> roomIDs) {
```